### PR TITLE
Issue #236: SuppressionPatchXpathFilter: OverloadMethodsDeclarationOrder's context strategy

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #236: SuppressionPatchXpathFilter: OverloadMethodsDeclarationOrder's context strategy

common pattern https://github.com/checkstyle/patch-filters/issues/8#issuecomment-665554887

### CODE REVIEW

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/OverloadMethodsDeclarationOrder/context/defaultContextConfig.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+          "https://checkstyle.org/dtds/configuration_1_3.dtd">
+<module name="Checker">
+    <module name="TreeWalker">
+        <module name="OverloadMethodsDeclarationOrder"/>
+
+        <module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchXpathFilter">
+            <property name="file" value="${tp}/defaultContext.patch" />
+            <property name="strategy" value="context" />
+            <property name="neverSuppressedChecks" value="OverloadMethodsDeclarationOrderCheck" />
```
Comment (rdiachenko) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/OverloadMethodsDeclarationOrder/context/defaultContextConfig.xml:12]: Why do you put it under this group? Can we use checkNamesForContextStrategyByTokenOrParentSet option instead?

Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/OverloadMethodsDeclarationOrder/context/defaultContextConfig.xml]: yes, it's ok, I use `checkNamesForContextStrategyByTokenOrParentSet` now.

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/OverloadMethodsDeclarationOrder/context/Test.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package TreeWalker.coding.OverloadMethodsDeclarationOrder;
+
+public class Test {
+    public void foo(int i) {} // OK
+    public void foo(String s) {} // OK
+    public void notFoo() {}  // violation without filter
```
Comment (rdiachenko) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/OverloadMethodsDeclarationOrder/context/Test.java:6]: Are you sure this comment is on the right place? The violation in the expected.txt shows line 7. Same for below cases.

Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/OverloadMethodsDeclarationOrder/context/Test.java]: sorry, I have corrected it.

**Pull request merged**